src/binary_tree/sparse_binary_tree.rs

```diff
@@ -0,0 +1,515 @@
+        &self.coord
+    }
+    pub fn get_x_coord(&self) -> u64 {
+        self.coord.x
+    }
+    pub fn get_y_coord(&self) -> u8 {
+        self.coord.y
+    }
+    pub fn get_content(&self) -> &C {
+        &self.content
+    }
+}
+
+impl<C: Clone> SparseBinaryTree<C> {
+    pub fn get_height(&self) -> u8 {
+        self.height
+    }
+    pub fn get_root(&self) -> &Node<C> {
+        &self.root
+    }
+    /// Attempt to find a Node via it's coordinate in the underlying store.
+    pub fn get_node(&self, coord: &Coordinate) -> Option<&Node<C>> {
+        self.store.get(coord)
+    }
+}
+
+// ===========================================
+// Supporting structs, types and functions.
+
+/// Used to organise nodes into left/right siblings.
+pub enum NodeOrientation {
+    Left,
+    Right,
+}
+
+impl<C: Clone> Node<C> {
+    /// Return true if self is a) a left sibling and b) lives just to the left of the other node.
+    #[allow(dead_code)]
+    pub fn is_left_sibling_of(&self, other: &Node<C>) -> bool {
+        match self.node_orientation() {
+            NodeOrientation::Left => {
+                self.coord.y == other.coord.y && self.coord.x + 1 == other.coord.x
+            }
+            NodeOrientation::Right => false,
+        }
+    }
+
+    /// Return true if self is a) a right sibling and b) lives just to the right of the other node.
+    pub fn is_right_sibling_of(&self, other: &Node<C>) -> bool {
+        match self.node_orientation() {
+            NodeOrientation::Left => false,
+            NodeOrientation::Right => {
+                self.coord.x > 0
+                    && self.coord.y == other.coord.y
+                    && self.coord.x - 1 == other.coord.x
+            }
+        }
+    }
+
+    /// Return the coordinates of this node's sibling, whether that be a right or a left sibling.
+    pub fn get_sibling_coord(&self) -> Coordinate {
+        match self.node_orientation() {
+            NodeOrientation::Left => Coordinate {
+                y: self.coord.y,
+                x: self.coord.x + 1,
+            },
+            NodeOrientation::Right => Coordinate {
+                y: self.coord.y,
+                x: self.coord.x - 1,
+            },
+        }
+    }
+
+    /// Return the coordinates of this node's parent.
+    /// The x-coord divide-by-2 works for both left _and_ right siblings because of truncation.
+    /// Note that this function can be misused if tree height is not used to bound the y-coord from above.
+    #[allow(dead_code)]
+    pub fn get_parent_coord(&self) -> Coordinate {
+        Coordinate {
+            y: self.coord.y + 1,
+            x: self.coord.x / 2,
+        }
+    }
+}
+
+impl<C: Clone> InputLeafNode<C> {
+    /// Convert the simpler node type to the actual Node type.
+    fn to_node(self) -> Node<C> {
+        Node {
+            content: self.content,
+            coord: Coordinate {
+                x: self.x_coord,
+                y: 0,
+            },
+        }
+    }
+}
+
```

Comment:
`LeftSibling<C>` and `RightSibling<C>` are both effectively `Node<C>`, so I would just say `Left(Node<C>)` and `Right(Node<C>)`